src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -52,11 +91,25 @@ public function validate($value, Constraint $constraint)
 
         $value = (string) $value;
 
-        if (null === $constraint->strict) {
-            $constraint->strict = $this->isStrict;
+        if (null !== $constraint->strict) {
+            @trigger_error(
```

Comment:
the `trigger_error()` call should be on a single line

Reply:
And IMO we should trigger this deprecation when the constraint is constructed to be able to catch it earlier than at runtime.

Reply:
I've added this to the constructor of the Email constraint class now. I am not sure if I should leave this in there as well or not.